updown/fetch.cgi

```diff
@@ -80,14 +80,65 @@ if ($ENV{'PATH_INFO'}) {
 		if (!$fetch_show) {
 			print "Content-Disposition: Attachment\n";
 			}
-		@st = stat($file);
-		print "Content-length: $st[7]\n";
-		print "X-Content-Type-Options: nosniff\n";
-		print "Content-type: $type\n\n";
-		while(read(FILE, $buffer, &get_buffer_size_binary())) {
-			print("$buffer");
+		# Stat file
+		my @st = stat($file);
+		my $fsize = $st[7];
+
+		# Get and analyze the file contents first
+		my $fdata = "";
+		my $dangertypes = $type =~ /html|xml|pdf/i;
+		my $htmltype    = $type =~ /html/i ? 1 : 0;
+		my $pdftype     = $type =~ /pdf/i ? 'pdf' : 0;
+		my $bsize =
+		    $dangertypes ? $fsize : &get_buffer_size_binary();
```

Comment:
>> [jcameron](https://github.com/jcameron) [3 hours ago](https://github.com/webmin/webmin/commit/3de93296a93c3843c7f6bc9f31c51996bd1de57b#r114170267)
>> It's not possible to read an entire file with the read() function .. instead use read_file_contents

---

> It's not possible to read an entire file with the read() function .. instead use read_file_contents

I started checking on it, to double check if it is possible to slurp the entire file using `read()` function .. and it appears to be possible .. read function in the third argument expects a _length_ of bytes to be read. If we pass in _length_ the actual file size, we read its full content in one go, isn't it right?

Also, [this article](https://www.perl.com/article/21/2013/4/21/Read-an-entire-file-into-a-string/) points out that it is also possible. Could you please double check?